src/diffusers/pipelines/consistency_models/pipeline_consistency_models.py

```diff
@@ -0,0 +1,312 @@
+from typing import Callable, List, Optional, Union
+
+import torch
+
+from ...models import UNet2DModel
+from ...schedulers import KarrasDiffusionSchedulers
+from ...utils import (
+    deprecate,
+    is_accelerate_available,
+    is_accelerate_version,
+    logging,
+    randn_tensor,
+)
+from ..pipeline_utils import DiffusionPipeline, ImagePipelineOutput
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+class ConsistencyModelPipeline(DiffusionPipeline):
+    r"""
+    Pipeline for consistency models for unconditional or class-conditional image generation, as introduced in [1].
+
+    This model inherits from [`DiffusionPipeline`]. Check the superclass documentation for the generic methods the
+    library implements for all the pipelines (such as downloading or saving, running on a particular device, etc.)
+
+    [1] Song, Yang and Dhariwal, Prafulla and Chen, Mark and Sutskever, Ilya. "Consistency Models"
+    https://arxiv.org/pdf/2303.01469
+
+    Args:
+        unet ([`UNet2DModel`]):
+            Unconditional or class-conditional U-Net architecture to denoise image latents.
+        scheduler ([`SchedulerMixin`]):
+            A scheduler to be used in combination with `unet` to denoise the image latents. Currently only compatible
+            with [`CMStochasticIterativeScheduler`].
+    """
+
+    def __init__(self, unet: UNet2DModel, scheduler: KarrasDiffusionSchedulers) -> None:
```

Comment:
Since we're specifying the type of scheduler to `KarrasDiffusionSchedulers`, have we checked if Karras-style schedulers work with the pipeline? If not, I think changing the type to be of `CMStochasticIterativeScheduler` makes sense. 

We could maybe even add a check at the beginning to throw an error if a user wants to load the pipeline with an incompatible scheduler. But I think it should not block the PR from getting merged.